### ISSUE TITLE
[TTS] AlignerModel bin_loss warmup ramp (bin_loss_scale) has no effect

### ISSUE THREAD

**Issue by udsy19** [NVIDIA-NeMo/NeMo#16170]
**Describe the bug**

`AlignerModel`'s bin-loss warmup ramp has no effect: `bin_loss` is always applied at full strength
from the first epoch it turns on, instead of ramping from 0 to 1 over `bin_loss_warmup_epochs`.

`nemo/collections/tts/models/aligner.py:135-144` (`on_train_epoch_start`) computes
`self.bin_loss_scale = min((self.current_epoch - bin_loss_start_epoch) / self.bin_loss_warmup_epochs, 1.0)`
every epoch once `add_bin_loss` turns on. But `_metrics` (`:123-133`) never reads it:

```python
if self.add_bin_loss:
    attn_hard = binarize_attention(attn_soft, text_len, spec_len)
    bin_loss = self.bin_loss(hard_attention=attn_hard, soft_attention=attn_soft)
    loss += bin_loss
```

`bin_loss_scale` has no other reader anywhere in the tree, so the warmup config knob
(`bin_loss_warmup_epochs`, shipped as `100` in `examples/tts/conf/aligner.yaml`) is dead: the moment
`current_epoch >= bin_loss_start_epoch`, the un-warmed hard-attention loss is added at full weight on
that very first epoch. The sibling `FastPitchModel` implements the identical ramp correctly
(`fastpitch.py:432-433`): `bin_loss_weight = min(self.current_epoch / self.bin_loss_warmup_epochs, 1.0)`,
then `bin_loss_fn(...) * bin_loss_weight`.

**Steps/Code to reproduce bug**

```python
import torch
from nemo.collections.tts.losses.aligner_loss import BinLoss, ForwardSumLoss
from nemo.collections.tts.models import AlignerModel

torch.manual_seed(0)
attn_soft = torch.softmax(torch.randn(1, 1, 6, 4), dim=-1)
attn_logprob = torch.log_softmax(torch.randn(1, 1, 6, 4), dim=-1)
spec_len, text_len = torch.tensor([6]), torch.tensor([4])

self = type('S', (), {})()
self.forward_sum_loss, self.bin_loss, self.add_bin_loss = ForwardSumLoss(), BinLoss(), True
self.bin_loss_scale = 0.0  # the ramp's own starting value
loss, forward_sum_loss, *_ = AlignerModel._metrics(self, attn_soft, attn_logprob, spec_len, text_len)
print(loss.item(), forward_sum_loss.item())  # should be equal at scale=0.0; they are not
```

On the current tip these two numbers differ (e.g. `3.063...` vs `1.536...`) — `bin_loss` is added in
full even though `bin_loss_scale` is `0.0`, the value the ramp holds it at for the whole warmup window.

**Expected behavior**

At `bin_loss_scale == 0.0` the total loss should equal `forward_sum_loss` alone (no `bin_loss`
contribution), ramping linearly up to the full `bin_loss` magnitude as `bin_loss_scale` reaches `1.0`.

**Environment overview**

- Environment location: bare metal
- Method of NeMo install: from source, `pip install -e .`

**Environment details**

- Ubuntu 24.04 · PyTorch 2.13.0+cpu · Python 3.12.3
